src/cmd/build.js

```diff
@@ -55,21 +61,35 @@ async function defaultPackageCreator(
   let buffer = await zipDir(sourceDir, {
     filter: (...args) => fileFilter.wantFile(...args),
   });
-  let messageData: any;
-  if (manifestData.default_locale) {
-    messageData = readFileSync(
-     path.join(sourceDir, '_locales', manifestData.default_locale,
-     'messages.json'));
+
+  async function getDefaultLocalizedName(
+  {manifestData, sourceDir}: LocalizedNameParams
+  ): Promise<string> {
+    let messageData: any;
+    let extensionName: string;
+
+    if (manifestData.default_locale) {
+      messageData = await fs.readFile(path.join(sourceDir, '_locales',
+                                    manifestData.default_locale,
+                                    'messages.json'));
+    }
+    extensionName = manifestData.name.replace(/__MSG_([A-Za-z0-9@_]+?)__/,
+                    function(match, messageName) {
+                      if (messageData[messageName]
```

Comment:
This branch is never true. My guess is that the value of messageData is still undefined when this line is executed. As a result, `match` which is `_msg_extensionname_` is being returned always & making the test fail.